File: src/main/java/com/github/simiacryptus/aicoder/openai/translate/TranslationRequest.java
package com.github.simiacryptus.aicoder.openai.translate;

import com.github.simiacryptus.aicoder.openai.CompletionRequest;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

public interface TranslationRequest {
    @NotNull CompletionRequest buildCompletionRequest();

    @SuppressWarnings("unused")
    String getInputTag();

    @SuppressWarnings("unused")
    String getOutputTag();

    @SuppressWarnings("unused")
    CharSequence getInstruction();

    @SuppressWarnings("unused")
    @NotNull Map<CharSequence, CharSequence> getInputAttr();

    @SuppressWarnings("unused")
    @NotNull Map<CharSequence, CharSequence> getOutputAttr();

    @SuppressWarnings("unused")
    String getOriginalText();

    @SuppressWarnings("unused")
    double getTemperature();

    @SuppressWarnings("unused")
    int getMaxTokens();

    TranslationRequest setInputType(CharSequence inputTag);

    TranslationRequest setOutputType(CharSequence outputTag);

    TranslationRequest setInstruction(CharSequence instruction);

    TranslationRequest setInputAttribute(CharSequence key, CharSequence value);
    TranslationRequest setOutputAttrute(CharSequence key, CharSequence value);

    TranslationRequest setInputText(CharSequence originalText);

    @SuppressWarnings("unused")
    TranslationRequest setTemperature(double temperature);

    @SuppressWarnings("unused")
    TranslationRequest setMaxTokens(int maxTokens);
}

File: src/main/kotlin/com/github/simiacryptus/aicoder/openai/translate/TranslationRequest.kt
package com.github.simiacryptus.aicoder.openai.translate

import com.github.simiacryptus.aicoder.openai.CompletionRequest
import java.util.*
import kotlin.collections.HashMap


interface TranslationRequest {
    fun buildCompletionRequest(): CompletionRequest

    @get:Suppress("unused")
    val inputTag: CharSequence?

    @get:Suppress("unused")
    val outputTag: CharSequence?

    @get:Suppress("unused")
    val instruction: CharSequence?

    @get:Suppress("unused")
    val inputAttr: Map<CharSequence?, CharSequence?>

    @get:Suppress("unused")
    val outputAttr: Map<CharSequence?, CharSequence?>

    @get:Suppress("unused")
    val originalText: CharSequence?

    @get:Suppress("unused")
    val temperature: Double

    @get:Suppress("unused")
    val maxTokens: Int

    fun setInputType(inputTag: CharSequence?): TranslationRequest
    fun setOutputType(outputTag: CharSequence?): TranslationRequest
    fun setInstruction(instruction: CharSequence?): TranslationRequest
    fun setInputAttribute(key: CharSequence?, value: CharSequence?): TranslationRequest
    fun setOutputAttrute(key: CharSequence?, value: CharSequence?): TranslationRequest
    fun addExample(exampleText: CharSequence, attributes: Map<CharSequence, CharSequence> = HashMap()): TranslationRequest

    fun setInputText(originalText: CharSequence?): TranslationRequest

    @Suppress("unused")
    fun setTemperature(temperature: Double): TranslationRequest

    @Suppress("unused")
    fun setMaxTokens(maxTokens: Int): TranslationRequest
}


